python/risk_meters/list_risk_scores.py

```diff
@@ -0,0 +1,99 @@
+
+    for risk_meter in risk_meters_resp:
+        risk_meter_id = risk_meter['id']
+        #query_string = risk_meter['querystring']
+        risk_meter_score = risk_meter['risk_meter_score']
+        true_score = risk_meter['true_risk_meter_score']
+        risk_meters[risk_meter['name']] = (risk_meter_id, risk_meter_score, true_score)
+
+    return risk_meters
+
+# Obtain and return the number of assets in a risk meter.
+if __name__ == "__main__":
+    print("")
+    print("List Risk Meter Scores")
+    print("~~~~~~~~~~~~~~~~~~~~~~")
+    print("")
+
+    # Obtain the Kenna Security API key from an environment variable.
+    api_key = os.getenv('KENNA_API_KEY')
+    if api_key is None:
+        print("API key is non-existent")
+        sys.exit(1)
+
+    # HTTP headers.
+    headers = {'X-Risk-Token': api_key,
+               'Content-Type': 'application/json; charset=utf-8',
+               'User-Agent': 'sample.list_risk_meter_scores/1.0.0 (Cisco Secure)'}
+
+    # You might have to change this depending on your deployment.
+    base_url = "https://api.kennasecurity.com/"
+
+    risk_meters = get_risk_meter_scores(base_url, headers)
+
+    # Initialize the output table.
+    risk_meter_tbl = PrettyTable()
+    risk_meter_tbl.field_names = ["Risk Meter Name", "ID", "Risk Score", "True Risk Score"]
+    risk_meter_tbl.align["Risk Meter Name"] = "l"
+
+    # Go through the list of risk meters filling out the output table.
+    for risk_meter_name in sorted (risk_meters.keys()):
+        risk_meter_tuple = risk_meters[risk_meter_name]
+        risk_meter_id = risk_meter_tuple[0]
+        risk_meter_score = risk_meter_tuple[1]
+        risk_meter_true_score = risk_meter_tuple[2]
+        true_score = f"{risk_meter_true_score}   " if risk_meter_true_score == risk_meter_score else f"{risk_meter_true_score} **"
+
+        risk_meter_tbl.add_row([risk_meter_name, risk_meter_id, risk_meter_score, true_score])
+
+    # Print the output table.
+    print(risk_meter_tbl)
+    print("")
+
+    print("Vulnerability Counts for Unequal Risk Scores")
+    print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+    print("")
+
+    # Go through the risk meters to output risk meter vulnerability counts for unequal scores.
+    for risk_meter_name in sorted (risk_meters.keys()):
+        risk_meter_tuple = risk_meters[risk_meter_name]
+        risk_meter_id = risk_meter_tuple[0]
+        risk_meter_score = risk_meter_tuple[1]
+        risk_meter_true_score = risk_meter_tuple[2]
+
+        if risk_meter_score != risk_meter_true_score:
+            rtn_code = 0
+            cmd = ["python",
+                   "historical_vuln_count.py",
+                   risk_meter_name,
+                   "today",
+                   str(risk_meter_id),
+                   str(risk_meter_score),
+                   str(risk_meter_true_score)]
+            subprocess.run(cmd)
+            subprocess.CompletedProcess(cmd, rtn_code)
```

Comment:
tl;dr I think this is appropriate use. 👍 

If it were very often that the true risk score would differ from risk score, then I'd worry that the spawning of the extra processes could be a performance hit if there are 100s or 1,000s of risk meters.  However, we shouldn't sweat this.  This is nice and clean, and it's really cool that historical_vuln_count.py can be invoked independently. ❤️

Reply:
Yeah, I thought about, but since it is only and example, I didn't sweat it.  A better option might have been to pass all the risks meters at one time.